### PR TITLE
Add list / download log file feature

### PR DESCRIPTION
**Pull request by GDLMadushanka**
Fixes wso2/micro-integrator/issues/1401

### CODE REVIEW

File: components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/Utils.java [hunk 1]
```diff
@@ -143,4 +149,89 @@ public static String getProperty(File srcFile, String key) throws IOException {
...
+        List<LogFileInfo> logFilesList = new ArrayList<>();
+        LogFileInfo logFileInfo;
+
+        File folder = new File(folderPath);
+        File[] listOfFiles = folder.listFiles();
+
+        if (listOfFiles == null || listOfFiles.length == 0) {
+            // folder.listFiles can return a null, in that case return a default log info
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("List of log files of the given pattern is null.");
```
Comment (sdkottegoda) [inline components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/Utils.java:202]: Let's print the log folder path here

File: components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/LogFilesResource.java [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+        ByteArrayDataSource bytArrayDS;
+        try {
+            Path logFilePath = Paths.get(Utils.getCarbonLogsPath(), logFile);
+            File file = new File(logFilePath.toString());
+            if (file.exists() && !file.isDirectory()) {
+                is = new BufferedInputStream(new FileInputStream(logFilePath.toString()));
+                bytArrayDS = new ByteArrayDataSource(is, "text/xml");
+                return new DataHandler(bytArrayDS);
+            } else {
+                log.error("Could not find the requested file : " + logFile);
```
Comment (sdkottegoda) [inline components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/LogFilesResource.java:157]: Shall we print the complete 'logFilePath'

File: components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/LogFilesResource.java [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+                return null;
+            }
+        } catch (IOException e) {
+            log.error("Error occurred while reading file : " + logFile, e);
```
Comment (sdkottegoda) [inline components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/LogFilesResource.java:161]: same here

File: components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/LogFilesResource.java [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+            return null;
+        } finally {
+            if (null != is) {
```
Comment (sdkottegoda) [inline components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/LogFilesResource.java:164]: We can use try with resources for this

File: components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/Utils.java (hunk 1 above)
Comment (GDLMadushanka) [reply components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/Utils.java]: +1 fixed

File: components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/LogFilesResource.java (hunk 2 above)
Comment (GDLMadushanka) [reply components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/LogFilesResource.java]: +1 fixed

File: components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/LogFilesResource.java (hunk 3 above)
Comment (GDLMadushanka) [reply components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/LogFilesResource.java]: +1 fixed

File: components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/LogFilesResource.java (hunk 4 above)
Comment (GDLMadushanka) [reply components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/LogFilesResource.java]: +1 fixed

**Pull request merged**